ckan/logic/action/update.py

```diff
@@ -320,9 +320,16 @@ def package_update(
         model.Session.rollback()
         raise ValidationError(errors)
 
-    #avoid revisioning by updating directly
+    # check for metadata_modified
+    if data_dict.get("metadata_modified"):
+        mmod = datetime.datetime.strptime(data_dict.get(
+            "metadata_modified"),  "%Y-%m-%dT%H:%M:%S.%f")
```

Comment:
instead of strptime here, put it in a validator on the field instead so that a validation error can be raised cleanly in the API.

Also `metadata_modified` will now need to be stripped any time data_dict is copied from existing data, e.g. in `package_patch`, `resource_create`, `resource_update` etc. Check for all existing callers to `package_update` to find other instances.